cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -466,80 +469,87 @@ ReconciliationState createReconciliationState(Reconciliation reconciliation, Kaf
         }
 
         /**
-         * If the SS exists, complete any pending rolls
+         * If the SS exists, check that the cluster is stable and no pods need rolling update
          *
-         * @return A Future which completes with the current state of the SS, or with null if the SS never existed.
+         * @return A Future which completes with the current quiescence state of the cluster, if it is stable or not
```

Comment:
Does it really still check whether it is stable? The changed method doesn't do readiness check anymore. So it will not know if it is stable or not.

Reply:
I had an out of band chat with @tombentley to understand what he meant as "check", it turned out that was just about checking generation and not readiness, maybe I miss understood but let's check with him.

Reply:
Well, I think this is not about what Tom meant but about what people will understand. You say _check that cluster is stable_ and I think that will mean for most people that the cluster is running and not crash looping. If you think that we do not need the readiness check here I'm fine with it. But the Javadoc is confusing then and should make it clear that it has nothing about stability but only about the generations and no need for rolling update.

Reply:
> I think this is not about what Tom meant ...
I don't agree here, because when someone else wrote some code I would take care of it and what was his rationale to do that in order to not break something.

Reply:
The Javadoc comment has to be understandable by everyone. Not just to the one who wrote the method. And for me _check that cluster is stable_ means that it is not crash looping. And that is not what the function does now.